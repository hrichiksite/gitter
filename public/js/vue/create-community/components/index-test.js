const mount = require('../../__test__/vuex-mount');
const { default: CreateCommunity } = require('./index.vue');
const appEvents = require('../../../utils/appevents');

import {
  CREATE_COMMUNITY_STEP_MAIN,
  CREATE_COMMUNITY_STEP_BACKING_ENTITY_GITLAB,
  CREATE_COMMUNITY_STEP_BACKING_ENTITY_GITHUB
} from '../constants';

describe('CreateCommunity', () => {
  it('main step matches snapshot', () => {
    const { wrapper } = mount(CreateCommunity, {}, store => {
      store.state.createCommunity.currentStep = CREATE_COMMUNITY_STEP_MAIN;
    });
    expect(wrapper.element).toMatchSnapshot();
  });

  it('backing entity GitLab matches snapshot', () => {
    const { wrapper } = mount(CreateCommunity, {}, store => {
      store.state.createCommunity.currentStep = CREATE_COMMUNITY_STEP_BACKING_ENTITY_GITLAB;
    });
    expect(wrapper.element).toMatchSnapshot();
  });

  it('backing entity GitHub matches snapshot', () => {
    const { wrapper } = mount(CreateCommunity, {}, store => {
      store.state.createCommunity.currentStep = CREATE_COMMUNITY_STEP_BACKING_ENTITY_GITHUB;
    });
    expect(wrapper.element).toMatchSnapshot();
  });

  it('fires off appEvents `destroy-create-community-view` event when close button is clicked', () => {
    const { wrapper } = mount(CreateCommunity);

    const destroyCreateCommunityViewEventFiredPromise = new Promise(resolve => {
      appEvents.on('destroy-create-community-view', () => {
        resolve();
      });
    });

    wrapper.find({ ref: 'closeButton' }).trigger('click');

    return expect(destroyCreateCommunityViewEventFiredPromise).resolves.toEqual();
  });
});
