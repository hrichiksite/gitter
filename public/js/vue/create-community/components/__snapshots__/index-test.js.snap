// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CreateCommunity backing entity GitHub matches snapshot 1`] = `
<div
  class="root js-create-community-view-root"
>
  <div
    class="container"
  >
    <button
      class="close-button"
    >
      <span
        class="close-icon"
      />
    </button>
     
    <!---->
     
    <!---->
     
    <step-backing-entity-stub
      orglist=""
      orgname="orgs"
      repolist=""
      reponame="repos"
    />
  </div>
</div>
`;

exports[`CreateCommunity backing entity GitLab matches snapshot 1`] = `
<div
  class="root js-create-community-view-root"
>
  <div
    class="container"
  >
    <button
      class="close-button"
    >
      <span
        class="close-icon"
      />
    </button>
     
    <!---->
     
    <step-backing-entity-stub
      orglist=""
      orgname="groups"
      repolist=""
      reponame="projects"
    />
     
    <!---->
  </div>
</div>
`;

exports[`CreateCommunity main step matches snapshot 1`] = `
<div
  class="root js-create-community-view-root"
>
  <div
    class="container"
  >
    <button
      class="close-button"
    >
      <span
        class="close-icon"
      />
    </button>
     
    <step-main-stub />
     
    <!---->
     
    <!---->
  </div>
</div>
`;
